### PR TITLE
feat(web2): added 'Advanced' tab for MTUs and Promiscuous mode

### PR DESCRIPTION
**Pull request by fdizazzo**
This PR adds an 'Advanced' tab in Kura Network Manager to configure MTUs for both IPv4 and IPv6 and Promiscuous mode configuration, as discussed in #5068 

The old MTU fields in Ip4 and Ip6 tab have been removed.

### CODE REVIEW

**Comment by mattdibi** [comment]
Hello there @fdizazzo 👋 

hope you had a great xmas. Just writing you here to inform you that I'll take care of the review ASAP, hopefully by the end of the week.

Cheers :)

**Comment by mattdibi** [comment]
Started by running some tests... there seems to be some validation issues

image

image

The happy path is working fine though

File: kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/client/ui/network/NetworkTabsUi.java
Comment (mattdibi) [inline kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/client/ui/network/NetworkTabsUi.java:1]: Please don't forget to update the copyright header

File: kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/client/ui/network/TabAdvancedUi.java [hunk 1]
```diff
@@ -0,0 +1,370 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Areti and others
```
Comment (mattdibi) [inline kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/client/ui/network/TabAdvancedUi.java]:
```suggestion
 * Copyright (c) 2024 Areti and others
```

File: kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/client/ui/network/TabAdvancedUi.ui.xml [hunk 2]
```diff
@@ -0,0 +1,93 @@
+<!DOCTYPE ui:UiBinder SYSTEM "http://dl.google.com/gwt/DTD/xhtml.ent">
+
+<!--
+
+    Copyright (c) 2023 Areti and/or its affiliates and others
```
Comment (mattdibi) [inline kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/client/ui/network/TabAdvancedUi.ui.xml]:
```suggestion
    Copyright (c) 2024 Areti and/or its affiliates and others
```

File: kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/client/ui/network/TabIp4Ui.java
Comment (mattdibi) [inline kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/client/ui/network/TabIp4Ui.java:1]: Please don't forget to update the copyright header

File: kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/client/ui/network/TabIp4Ui.ui.xml
Comment (mattdibi) [inline kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/client/ui/network/TabIp4Ui.ui.xml:1]: Please don't forget to update the copyright header

File: kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/client/ui/network/TabIp6Ui.java
Comment (mattdibi) [inline kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/client/ui/network/TabIp6Ui.java:1]: Please don't forget to update the copyright header

File: kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/client/ui/network/TabIp6Ui.ui.xml
Comment (mattdibi) [inline kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/client/ui/network/TabIp6Ui.ui.xml:1]: Please don't forget to update the copyright header

File: kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/server/net2/configuration/GwtNetInterfaceConfigBuilder.java
Comment (mattdibi) [inline kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/server/net2/configuration/GwtNetInterfaceConfigBuilder.java:1]: Please don't forget to update the copyright header

File: kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/server/net2/configuration/NetworkConfigurationServiceProperties.java
Comment (mattdibi) [inline kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/server/net2/configuration/NetworkConfigurationServiceProperties.java:1]: Please don't forget to update the copyright header

File: kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/server/net2/configuration/NetworkConfigurationServicePropertiesBuilder.java
Comment (mattdibi) [inline kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/server/net2/configuration/NetworkConfigurationServicePropertiesBuilder.java:1]: Please don't forget to update the copyright header

File: kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/shared/model/GwtNetInterfaceConfig.java
Comment (mattdibi) [inline kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/shared/model/GwtNetInterfaceConfig.java:1]: Please don't forget to update the copyright header

File: kura/org.eclipse.kura.web2/src/main/resources/org/eclipse/kura/web/client/messages/Messages.properties
Comment (mattdibi) [inline kura/org.eclipse.kura.web2/src/main/resources/org/eclipse/kura/web/client/messages/Messages.properties:1]: Please don't forget to update the copyright header

File: kura/org.eclipse.kura.web2/src/main/resources/org/eclipse/kura/web/client/messages/Messages.properties [hunk 3]
```diff
@@ -338,7 +338,7 @@ netIPv4ToolTipSubnetMask=For static configuration, enter the subnet mask that wi
 netIPv4ToolTipGateway=For static configuration, enter the default gateway for the device.
 netIPv4ToolTipRenew=For DHCP configuration, clicking this button will release the current IP address and acquire a new lease from the server.
 netIPv4ToolTipDns=List of DNS servers.<br/>New servers can be added by entering an address and clicking apply.<br/>To remove a server, delete the address from the list and click apply.<br/>Servers MUST be separated by a comma (,).<br/><br/>When configured for DHCP, manually entered DNS entries will replace the ones provided by the DHCP server.  Removing them will restore the DHCP entries.
-netIPv4Mtu=MTU
+netIPv4Mtu=IPv4 MTU
 netIPv4InvalidMtu=MTU must be an Integer.
 netIPv4ToolTipMtu=Selects the Maximum Transition Unit (MTU) for IPv4 traffic over this interface.<br/> A value of 0 allows MTU auto-discovery.
```
Comment (mattdibi) [inline kura/org.eclipse.kura.web2/src/main/resources/org/eclipse/kura/web/client/messages/Messages.properties]: Maybe it's better to group all the "Advanced" tab parameter together?

File: kura/org.eclipse.kura.web2/src/main/resources/org/eclipse/kura/web/client/messages/Messages.properties [hunk 4]
```diff
@@ -356,7 +356,7 @@ netIPv6SubnetMask=Subnet Mask
 netIPv6Gateway=Gateway
 netIPv6DNSServers=DNS Servers
 netIPv6Privacy=Privacy
-netIPv6Mtu=MTU
+netIPv6Mtu=IPv6 MTU
```
Comment (mattdibi) [inline kura/org.eclipse.kura.web2/src/main/resources/org/eclipse/kura/web/client/messages/Messages.properties]: Maybe it's better to group all the "Advanced" tab parameter together?

File: kura/org.eclipse.kura.web2/src/main/resources/org/eclipse/kura/web/client/messages/ValidationMessages.properties
Comment (mattdibi) [inline kura/org.eclipse.kura.web2/src/main/resources/org/eclipse/kura/web/client/messages/ValidationMessages.properties:1]: Please don't forget to update the copyright header

**Review by mattdibi — CHANGES_REQUESTED** [review]

**Comment by mattdibi** [comment]
Also... @marcellorinaldo I would like to ask you to give this PR a look since you worked extensively on the new networking UI

File: kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/client/ui/network/TabIp6Ui.ui.xml [hunk 5]
```diff
@@ -12,6 +12,7 @@
 	
 	Contributors:
 	 Eurotech
+     Areti
```
Comment (mattdibi) [inline kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/client/ui/network/TabIp6Ui.ui.xml]: Some non-printable character issue here:

```suggestion
	 Areti
```

**Review by mattdibi — APPROVED** [review]
LGTM. Validation issues were fixed. 👍

**Review by marcellorinaldo — COMMENTED** [review]
We should open a documentation PR that updates the "Network Configuration" section specifying that the advanced tab is available only on generic profiles.

File: kura/org.eclipse.kura.web2/src/main/resources/org/eclipse/kura/web/client/messages/Messages.properties [hunk 6]
```diff
@@ -376,7 +372,16 @@ netIPv6ToolTipSubnetMask=For static configuration, enter the subnet mask that wi
 netIPv6ToolTipGateway=For static configuration, enter the IPv6 address of the default gateway for the device.
 netIPv6ToolTipDns=List of DNS servers.<br><br>Servers MUST be separated by a space or a newline.<br><br>When configured for DHCP, manually entered DNS entries will replace the ones provided by the DHCP server. Removing them will restore the DHCP entries.
 netIPv6ToolTipPrivacy=Configure IPv6 Privacy Extensions for SLAAC, described in RFC4941. If enabled, it makes the kernel generate a temporary IPv6 address in addition to the public one generated from MAC address via modified EUI-64. Note that this setting is distinct from the Stable Privacy addresses that can be enabled with the Address Generation Mode property as another way of avoiding host tracking with IPv6 addresses
-netIPv6ToolTipMtu=Selects the Maximum Transition Unit (MTU) for IPv6 traffic over this interface.<br/> A value of 0 allows MTU auto-discovery.<br/>Requires NetworkManager 1.40 or newer.
+
+netAdvanced=Advanced
```
Comment (marcellorinaldo) [inline kura/org.eclipse.kura.web2/src/main/resources/org/eclipse/kura/web/client/messages/Messages.properties:376]: Please update also the Japanese translation: https://github.com/eclipse/kura/blob/develop/kura/org.eclipse.kura.web2/src/main/resources/org/eclipse/kura/web/client/messages/Messages_ja.properties

Just copy there the Google translate translation, I will make it review by one of our collegues later.

**Review by marcellorinaldo — COMMENTED** [review]
Other than the points above, the PR seems good to me. I will approve when the documentation PR is opened and the Japanese translation is set.

Many thanks @fdizazzo

File: kura/org.eclipse.kura.web2/src/main/resources/org/eclipse/kura/web/client/messages/Messages.properties (hunk 6 above)
Comment (fdizazzo) [reply kura/org.eclipse.kura.web2/src/main/resources/org/eclipse/kura/web/client/messages/Messages.properties:376]: Just pushed the japanese labels, will add the docs pr in the next few days.

> the "Network Configuration" section specifying that the advanced tab is available only on generic profiles

Is it specifically generic profiles or only on profiles using Kura's new network management?

Comment (marcellorinaldo) [reply kura/org.eclipse.kura.web2/src/main/resources/org/eclipse/kura/web/client/messages/Messages.properties:376]: I would stick with just "generic profiles", as here: http://eclipse.github.io/kura/docs-release-5.4/getting-started/install-kura/

**Review by marcellorinaldo — APPROVED** [review]

**Pull request merged**